### DISCUSSION TITLE
Please help, I can't download plugins over proxy.

### DISCUSSION THREAD

**Discussion by gerroon** [VSCodium/vscodium#1762] [Q&A]
Hi

I have proxy set in the settings and I can't download p;plugins for some reason.  Now I am also behind firewall, I already enabled Vscodium exe to access the internet but still hitting an issue.

I tried both the open plugin servers and the MS market place still no luck. I am wondering if anyone can give me hints?  I can use internet connected plugins like AWS. So definitely not a direct connection issue.

```
    "http.proxyStrictSSL":false,
    "http.proxy": "http://192.168.1.10:8118",
    "https.proxy": "http://192.168.1.10:8118",
```

```
	at process.callbackTrampoline (node:internal/async_hooks:128:24)
2023-12-11 22:52:06.760 [error] jedi.api.environment.InvalidPythonEnvironment: Scripts\python.exe seems to be missing.: Error: jedi.api.environment.InvalidPythonEnvironment: Scripts\python.exe seems to be missing.
	at c:\VSCODE\.vscode-oss\extensions\ms-python.python-2023.23.13391009\out\client\extension.js:2:2338987
	at re (c:\VSCODE\.vscode-oss\extensions\ms-python.python-2023.23.13391009\out\client\extension.js:2:2339281)
	at c:\VSCODE\.vscode-oss\extensions\ms-python.python-2023.23.13391009\out\client\extension.js:2:2334065
	at Immediate.<anonymous> (c:\VSCODE\.vscode-oss\extensions\ms-python.python-2023.23.13391009\out\client\extension.js:2:2334085)
	at process.processImmediate (node:internal/timers:476:21)
	at process.topLevelDomainCallback (node:domain:161:15)
	at process.callbackTrampoline (node:internal/async_hooks:128:24)
2023-12-12 11:04:37.123 [error] Error: net::ERR_CONNECTION_REFUSED
    at SimpleURLLoaderWrapper.<anonymous> (node:electron/js2c/browser_init:2:49935)
    at SimpleURLLoaderWrapper.emit (node:events:513:28)
2023-12-12 11:04:37.141 [error] net::ERR_CONNECTION_REFUSED: Error: net::ERR_CONNECTION_REFUSED
    at SimpleURLLoaderWrapper.<anonymous> (node:electron/js2c/browser_init:2:49935)
    at SimpleURLLoaderWrapper.emit (node:events:513:28)


```

**Comment by Jeong-ilKajokaya** [comment]
Hey there,

I see what’s happening with your plugin downloads. It looks like even though your proxy is set up, VSCodium is having trouble connecting to the servers. This is pretty common behind firewalls.

A few things you can try:

### 1. Double-check your proxy settings

Make sure they match the environment variables that Node/Electron will actually see:

```bash
set HTTP_PROXY=http://192.168.1.10:8118
set HTTPS_PROXY=http://192.168.1.10:8118
```

On Linux/macOS, use `export` instead of `set`.

Also, your config with `http.proxyStrictSSL: false` is correct for self-signed certificates, so that’s fine.

### 2. Use the `--proxy-server` flag

Sometimes Electron apps need this explicitly:

```bash
codium --proxy-server=http://192.168.1.10:8118
```

### 3. Test outside VSCodium

From the same machine, can you `curl` or `wget` the URLs of the extensions? For example:

```bash
curl -v https://open-vsx.org/api
```

If that fails, it’s a network-level block, not VSCodium.

### 4. Python extension errors

Your logs also show Python extension errors (`jedi.api.environment.InvalidPythonEnvironment`). That’s unrelated to proxy, it just means the Python interpreter path isn’t valid. Make sure your Python install is accessible and correctly set in VSCodium settings.

### 5. Last resort

If the network is tricky, you can manually download `.vsix` files for extensions and install them locally:

```bash
Extensions -> Install from VSIX...
```

This way you can bypass the proxy issues entirely.

**TL;DR:** The proxy looks fine in settings, but Electron sometimes needs environment variables or the `--proxy-server` flag. Test with `curl` to confirm connectivity, and for Python issues, make sure your interpreter path is correct.
